packages/scm/src/browser/scm-widget.tsx

```diff
@@ -87,6 +87,7 @@ export class ScmWidget extends ReactWidget implements StatefulWidget {
     protected init(): void {
         this.refresh();
         this.toDispose.push(this.scmService.onDidChangeSelectedRepository(() => this.refresh()));
+        this.toDispose.push(this.labelProvider.onDidChange(() => this.refresh()));
```

Comment:
it should be `this.update` to avoid rescanning repositories, but only rerender the widget